src/main/java/seedu/address/storage/JsonAdaptedSchedule.java

```diff
@@ -0,0 +1,54 @@
+package seedu.address.storage;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.commons.exceptions.InvalidEnumArgumentException;
+import seedu.address.model.schedule.Event;
+import seedu.address.model.schedule.Schedule;
+
+public class JsonAdaptedSchedule {
+    private final List<JsonAdaptedEvent> events = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonAdaptedSchedule} with the given schedule details.
+     */
+    @JsonCreator
+    public JsonAdaptedSchedule(@JsonProperty("schedule") List<JsonAdaptedEvent> events) {
+        if (events != null) {
+            this.events.addAll(events);
+        }
+    }
+
+    /**
+     * Converts a given {@code Schedule} into this class for Jackson use.
+     */
+    public JsonAdaptedSchedule(Schedule source) {
+        events.addAll(source.getEvents().stream()
+                .map(JsonAdaptedEvent::new)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted schedule object into the model's {@code Schedule} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted schedule.
+     * @throws InvalidEnumArgumentException if there are any unhandled RecurFrequency cases.
+     */
+    public Schedule toModelType() throws IllegalValueException, InvalidEnumArgumentException {
+        final List<Event> schedule = new ArrayList<>();
```

Comment:
While we are using schedule to contain a list of events, I don't think we should call it schedule here.
```suggestion
        final List<Event> toAddEvents = new ArrayList<>();
```

Reply:
I was looking at this code again, would it be better if I renamed it modelEvents instead? So that it is consistent with the other JsonAdapted classes.

Reply:
Yea. Sounds good.